src/fhirtypes/common.ts

```diff
@@ -271,7 +267,7 @@ export function replaceReferences(
  * @param rule - The rule to get a value from
  * @returns - The value on the rule
  */
-function getRuleValue(rule: FixedValueRule | CaretValueRule): FixedValueType {
+function getRuleValue(rule: any): FixedValueType {
```

Comment:
When I tested this locally, I did not have to change this to `any`.  I think I remember it working w/ `FixedValueRule | CaretValueRule` just fine.  Did you *have* to change this?

Reply:
I get:
```
Argument of type 'T' is not assignable to parameter of type 'FixedValueRule | CaretValueRule'.
  Type 'T' is not assignable to type 'CaretValueRule'
```
when I change it, which seems sensible right? Becuase TypeScript doesn't know what `T` is, and can't be sure it'll work for a FixedValueRule or CaretValueRule.

Reply:
Oh, I didn't notice... you didn't constrain T at all.  It should be fine if you do this instead:
```ts
export function replaceReferences<T extends FixedValueRule | CaretValueRule>(rule: T, tank: FSHTank, fisher: Fishable): T {
```

Reply:
Ohhhh that makes more sense